src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,363 @@
+        Point ball_contact_point =
+            robot.position() + Vector::createFromAngle(robot.orientation())
+                                   .normalize(dist_to_ball_in_dribbler);
+
+        // Find the closest point to the ball contact point on the ball's trajectory
+        Point closest_ball_pos = ball.position();
+        if (ball.velocity().length() >= BALL_MIN_MOVEMENT_SPEED)
+        {
+            closest_ball_pos =
+                closestPoint(ball_contact_point,
+                             Line(ball.position(), ball.position() + ball.velocity()));
+        }
+        Ray shot(closest_ball_pos, best_shot_target - closest_ball_pos);
+
+        Angle ideal_orientation      = getOneTouchShotDirection(shot, ball);
+        Vector ideal_orientation_vec = Vector::createFromAngle(ideal_orientation);
+
+        // The best position is determined such that the robot stays in the ideal
+        // orientation, but moves the shortest distance possible to put its contact
+        // point in the ball's path.
+        Point ideal_position =
+            closest_ball_pos - ideal_orientation_vec.normalize(dist_to_ball_in_dribbler);
+
+        return Shot(ideal_position, ideal_orientation);
+    }
+
+    /*
+     * Finds a shot that is greater than MIN_SHOT_NET_PERCENT_OPEN and
+     * respects MAX_DEFLECTION_FOR_ONE_TOUCH_SHOT for the highest chance
+     * of scoring with a one-touch shot. If neither of those are true, return a nullopt
+     *
+     * @param world The world to find a feasible shot on
+     * @param assigned_robot The robot that will be performing the one-touch
+     */
+    static std::optional<Shot> findFeasibleShot(const World& world,
+                                                const Robot& assigned_robot)
+    {
+        // Check if we can shoot on the enemy goal from the receiver position
+        std::optional<Shot> best_shot_opt = calcBestShotOnGoal(
+            world.field(), world.friendlyTeam(), world.enemyTeam(),
+            assigned_robot.position(), TeamType::ENEMY, {assigned_robot});
+
+        // The percentage of open net the robot would shoot on
+        if (best_shot_opt)
+        {
+            // Vector from the ball to the robot
+            Vector robot_to_ball = world.ball().position() - assigned_robot.position();
+
+            // The angle the robot will have to deflect the ball to shoot
+            Angle abs_angle_between_pass_and_shot_vectors;
+
+            Vector robot_to_shot_target =
+                best_shot_opt->getPointToShootAt() - assigned_robot.position();
+            abs_angle_between_pass_and_shot_vectors =
+                (robot_to_ball.orientation() - robot_to_shot_target.orientation())
+                    .clamp()
+                    .abs();
+
+            Angle goal_angle =
+                acuteAngle(world.field().friendlyGoalpostPos(), assigned_robot.position(),
+                           world.field().friendlyGoalpostNeg())
+                    .abs();
```

Comment:
no need for abs since acuteAngle is already abs

Reply:
idk why this is so inconsistent lol, I blame it on robocup

fixed